### PR TITLE
FDWrapper: Do not try to close negative file descriptors

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

It turns out that some of the BPF helper functions return a negative `errno` value in case of failure, and since we wrap the return value into a `FDWrapper` right away this led to a warning from Valgrind about trying to close an invalid file descriptor.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Review by omoerbeek — APPROVED** [review]

File: pdns/misc.hh [hunk 1]
```diff
@@ -803,7 +803,7 @@ struct FDWrapper
 
   FDWrapper& operator=(FDWrapper&& rhs) noexcept
   {
-    if (d_fd != -1) {
+    if (d_fd >= 0) {
```
Comment (cmouse) [inline pdns/misc.hh:806]: should there be check here that fd != -1?

File: pdns/misc.hh [hunk 2]
```diff
@@ -824,7 +824,7 @@ struct FDWrapper
   int reset()
   {
     int ret = 0;
-    if (d_fd != -1) {
+    if (d_fd >= 0) {
       ret = close(d_fd);
       d_fd = -1;
```
[the remark this answers is not in the document]
Comment (rgacogne) [reply pdns/misc.hh:829]: Will do!

File: pdns/misc.hh (hunk 1 above)
Comment (rgacogne) [reply pdns/misc.hh:806]: I'm not sure I get what you have in mind?

Comment (cmouse) [reply pdns/misc.hh:806]: i was just wondering if it is acceptable, or not, to pass around closed fd handle?

Comment (rgacogne) [reply pdns/misc.hh:806]: Ah, right! I believe it is fine, yes, so I don't think we need an additional check.

**Review by cmouse — APPROVED** [review]

**Pull request merged**